test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchReaderTests.cs

```diff
@@ -363,6 +363,206 @@ public async Task ReadBatchRequestWithChangesetAndDependsOnIdsAsync()
                 isResponse: false);
         }
 
+        [Fact]
+        public void ReadBatchRequestWithNullHeaders()
+        {
+            var payload = "{\"requests\": [{" +
+                "\"id\": \"1\"," +
+                "\"method\": \"POST\"," +
+                "\"url\": \"http://tempuri.org/Customers\"," +
+                "\"headers\": {\"odata-version\":\"4.0\",\"content-type\":\"application/json;odata.metadata=minimal;odata.streaming=true;IEEE754Compatible=false;charset=utf-8\",\"null-header\":null}, " +
+                "\"body\": {\"@odata.type\":\"#NS.Customer\",\"Id\":1,\"Name\":\"Customer 1\",\"Type\":\"Retail\"}}]}";
+
+            SetupJsonLightBatchReaderAndRunTest(
+                payload,
+                (jsonLightBatchReader) =>
+                {
+                    try
+                    {
+                        while (jsonLightBatchReader.Read())
+                        {
+                            if (jsonLightBatchReader.State == ODataBatchReaderState.Operation)
+                            {
+                                var operationRequestMessage = jsonLightBatchReader.CreateOperationRequestMessage();
+                                // Verify that the Property "null-header" exists and it's value is set to NULL
+                                var nullHeaderProperty = operationRequestMessage.Headers.FirstOrDefault(p => p.Key == "null-header");
+                                Assert.NotNull(nullHeaderProperty.Key);
+                                Assert.Null(nullHeaderProperty.Value);
+                            }
+                        }
+                    }
+                    catch (NullReferenceException ex)

```

Comment:
A scenario where the library ends up throwing a `NullReferenceException` is not acceptable. That's bad practice. If the header is required, we should check if it's null and throw an `ODataException` explaining that a required header is missing.

Meaning that instead of
```csharp
string headerValue = this.jsonReader.ReadPrimitiveValue()?.ToString();
```
we should have something like:
```csharp
string headerValue = this.jsonReader.ReadPrimitiveValue()?.ToString();
if (headerValue == null)
{
    throw new ODataException("###ERROR_MESSAGE###");
}
```
or
```csharp
object headerObjValue = this.jsonReader.ReadPrimitiveValue();
if (headerObjValue == null)
{
    throw new ODataException("###ERROR_MESSAGE###");
}

string headerValue = headerObjValue.ToString()
```

Reply:
@gathogojr do you understand the purpose of this unit test?

I made this unit test to proof that no NullReferenceException is throw, by fixing the behavior of the JSON property reader.
If you are telling me, that you don't want me to handle JSON properties with `null` values, I suggest you to change the OData Protocol specifications and then handle it in the way you want it to.

I've done a similar on request of Mike for handling duplicate headers, because he insisted to change the OData specifications to not allow any duplicate headers.
But I won't do this a 2nd time here. It is kind of hard to believe, that this OData project is in productive usage by customers, without having proper unit test coverage, without having basic null or duplicate value validation, and every person doing review here, having a different idea on how the application has to operate and asking for the specification to change.

I'll do as following, revert my changes, removing the unit tests for NullReferenceExceptions check and my adjustments of valid handling of JSON properties, that include `null` values and leave it up to you, to decide and handle, how these should be handled. You can do as you like, either throw an OData Exception or handle these values correctly, it is now up to you to decide and do.

Reply:
@apiaskowski Reverting the changes was unnecessary. Explaining that the particular test verifies that a `NullReferenceException` is NOT thrown was sufficient. That was a misunderstanding on my part and I apologize for that.

Reply:
Sure, no issue, I can add the null value validation(s) as part of a separate PR (if and when decided on how to handle these).
I'd like to get these changes here reviewed, done and delivered.
Adding/improving code coverage, changing specifications, etc. we can always do in separate steps and don't need to block bug fixes.